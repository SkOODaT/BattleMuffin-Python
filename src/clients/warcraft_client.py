import requests
from authlib.integrations.requests_client import OAuth2Session
from uplink import Consumer, Query, get, params, returns

from config.client_config import ClientConfiguration
from config.region_config import Locale, Region


class WarcraftClient(Consumer):
    def __init__(self, client_id: str, client_secret: str, region: Region = Region.us, locale: Locale = Locale.en_US):
        self.client_config = ClientConfiguration(client_id, client_secret, region, locale)

        oath_response = requests.get(f"{self.client_config.oauth_host}/oauth/.well-known/openid-configuration").json()
        token_endpoint = oath_response['token_endpoint']
        session = OAuth2Session(client_id, client_secret, scope="openid", token_endpoint=token_endpoint, grant_type="client_credentials")
        session.fetch_token()
        super().__init__(base_url=self.client_config.host, client=session)
        self._inject(Query("locale").with_value(self.client_config.locale.name))

    @returns.json
    @params({"namespace": "static-us"})
    @get("/data/wow/achievement-category/index")
    def get_achievement_category_index(self):
        """Returns an index of achievement categories."""
