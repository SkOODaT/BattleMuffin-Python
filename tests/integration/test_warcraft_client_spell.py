import os

from battlemuffin.clients.warcraft_client import WarcraftClient
from battlemuffin.config.region_config import Locale, Region


def test_get_spell(snapshot):
    client = WarcraftClient(
        os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"), Region.us, Locale.en_US
    )
    response = client.get_spell(196607)
    assert response == snapshot


def test_get_spell_media(snapshot):
    client = WarcraftClient(
        os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"), Region.us, Locale.en_US
    )
    response = client.get_spell_media(196607)
    assert response == snapshot
