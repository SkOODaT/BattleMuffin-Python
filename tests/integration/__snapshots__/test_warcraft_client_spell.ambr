# name: test_get_spell
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/spell/196607?namespace=static-8.3.0_32861-us',
      },
    },
    'description': 'Tiger Palm also applies Eye of the Tiger, dealing 168 Nature damage to the enemy and 168 healing to the Monk over 8 sec.',
    'id': 196607,
    'media': <class 'dict'> {
      'id': 196607,
      'key': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/spell/196607?namespace=static-8.3.0_32861-us',
      },
    },
    'name': 'Eye of the Tiger',
  }
---
# name: test_get_spell_media
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/spell/196607?namespace=static-8.3.0_32861-us',
      },
    },
    'assets': <class 'list'> [
      <class 'dict'> {
        'key': 'icon',
        'value': 'https://render-us.worldofwarcraft.com/icons/56/ability_druid_primalprecision.jpg',
      },
    ],
    'id': 196607,
  }
---
